### ISSUE TITLE
Rerouting issue - reference to #760

### ISSUE THREAD

**Issue by Niduank** [mapbox/mapbox-navigation-android#1011]
@danesfeder

I have again a reroute issue while using MapboxNavigation.

app.gradle
```java
implementation 'com.mapbox.mapboxsdk:mapbox-android-sdk:6.1.3'
implementation 'com.mapbox.mapboxsdk:mapbox-android-core:0.2.0'
implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation:0.14.0'
implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation-ui:0.14.0'
implementation 'com.mapbox.mapboxsdk:mapbox-android-plugin-locationlayer:0.6.0-SNAPSHOT'
```

Coordinates to test it :
departure : 48.888584,2.247150
arrival : 48.8555654,2.3589835

Test simulation video : 
https://vimeo.com/274510385

**Comment by danesfeder** [comment]
@Niduank would you mind testing with `0.20.0`?  We've made some major improvements to our route-following logic.  Thanks!

**Comment by danesfeder** [comment]
Hey @Niduank, I'm going to close this as resolved with `0.20.0`, please feel free to re-open if you see the issue persists.  Also noting our latest release is `0.21.0`
